src/main/java/com/synopsys/integration/alert/workflow/scheduled/update/UpdateChecker.java

```diff
@@ -112,31 +128,83 @@ private boolean isComprisedOfNumericTokens(final String version) {
                    .allMatch(NumberUtils::isParsable);
     }
 
-    private Comparator<String> versionOrder() {
-        return (firstVersion, secondVersion) -> {
-            final String[] firstVersionTokens = StringUtils.split(firstVersion, VERSION_SEPARATOR);
-            final String[] secondVersionTokens = StringUtils.split(secondVersion, VERSION_SEPARATOR);
+    private Comparator<DockerTagModel> tagOrder() {
```

Comment:
A separate SemanticVersion model class (Or perhaps an already built utility) would probably make this all easier to follow. It would separate out version comparisons from this task.